src/components/Categories/Categories.js

```diff
@@ -0,0 +1,78 @@
+import {useEffect, useState} from 'react';
+import styled from 'styled-components';
+
+export default function Categories({menus, handleCategoryClick, curClickedCategory}) {
+  const [categories, setCategories] = useState([]);
+
+  useEffect(() => {
+    /* Get categories form the given menus */
+    function getCategories(menus) {
+      const categories = menus.reduce(
+        function (pre, cur) {
+          if (!pre.includes(cur.category)) {
+            pre.push(cur.category);
+          }
+          return pre;
+        },
+        ['All']
+      );
+      return categories;
+    }
+
+    setCategories(getCategories(menus));
+
+    /* Sort categories in this order */
+    const sortBy = ['All', 'Starters', 'Mains', 'Pastries', 'Drinks'];
+    const categorySort = ({data, sortBy, sortfield}) => {
+      const sortByObj = sortBy.reduce(
+        (obj, item, index) => ({
+          ...obj,
+          [item]: index,
+        }),
+        {}
+      );
+      return data.sort((a, b) => sortByObj[a[sortfield]] - sortByObj[b[sortfield]]);
+    };
+
+    categorySort({data: menus, sortBy, sortfield: 'category'});
+    /* eslint-disable-next-line */
```

Comment:
Please delete this eslint-disable. Try to fix the error instead.

Reply:
Yes, maybe you can help me out, useEffect need a missing dependency: menus, but if I add it to dependency array, it's also wrong! I receive another error. Add also I've found out, in some website, it's recommended to commit this out.

Reply:
As I can remenber Jerry has once also said we can commit it out. It should be a common problem, what do the others do under this situation?

Reply:
Okay Gregor another headcoach told me it's not recommend to disable eslint in this case.

Reply:
> 

Jerry suggests to post this in question channel. Thank you anyway to bring this up, I also want to get a clear question about this.